crates/deploy/src/lib.rs

```diff
@@ -2,51 +2,174 @@ pub mod deploy_proto {
     include!("../proto/ambient.run.deploy.rs");
 }
 
-use std::{path::Path, str::FromStr};
+use std::{
+    collections::HashMap,
+    path::{Path, PathBuf},
+    str::FromStr,
+};
 
 use ambient_project::Manifest;
+use md5::Digest;
```

Comment:
Why md5 rather than sha256?

I may be wrong but isn't MD5 considered cryptographically insecure as collisions can be fabricated using normal computers, which would set us up for DoS, though I may just be concerned in vain.

Reply:
@ten3roberts You need to consider where it's used as well; this is for hashing asset files for fingerprints, not for cryptography.

Reply:
Of course, though wouldn't a man in the middle be possible since you can send a different fabricated file that has the same MD5 sum, or mischievously replace things.

Do I think that is not a large concern of us as we don't use it for security related functionality or banking.

I just wanted to shed light on the possibility just so we at least had a discussion and evaluation of it.

Reply:
MD5 mostly because it's already used by Google Cloud Storage that we use in the backend. It calculates it for us when a file is uploaded.

As for collisions it shouldn't be a problem (basically what @FredrikNoren said).

Reply:
@ten3roberts Same as the discussion in the channel; we need to prioritize what we put our energy into. Right now three of us are putting time and energy into discussing this problem; which means we're not working on something more important. It's important to make sure we only start discussions that are actually important, otherwise we drown in the noise (if we just discuss everything). The problem we're discussing here is clearly extremely hypothetical, and if we listed all of the things we need to do to make ambient work it would come extremely far down; so we clearly should not put time and effort into working on it (by discussing it).